GitUI/UserControls/RevisionGrid/RevisionGridControl.cs

```diff
@@ -914,16 +913,46 @@ public void PerformRefreshRevisions(Func<RefsFilter, IReadOnlyList<IGitRef>> get
                 _gridView.SelectionChanged += OnGridViewSelectionChanged;
                 _gridView.ResumeLayout();
 
+                // Add back and show the spinner control, which was removed by SetPage call
+                Controls.Add(_loadingControlSync);
+                ShowLoading();
+
+                cancellationToken.ThrowIfCancellationRequested();
+
+                // Find all ambiguous refs (including stash, notes etc)
+                // Note: GetRefs can be very slow operation for large repos, start in parallel to git-log.
+                // refsByObjectId is needed when first output from git-log is handled
+                Func<IReadOnlyList<IGitRef>> refs = () => (getRefs ?? Module.GetRefs)(RefsFilter.NoFilter);
```

Comment:
```suggestion
                Func<IReadOnlyList<IGitRef>> refsFactory = () => (getRefs ?? Module.GetRefs)(RefsFilter.NoFilter);
```
or `getUnfilteredRefs`

Reply:
getUnfilteredRefs